wall_e/src/WalleModels/models.py

```diff
@@ -1,11 +1,74 @@
+# for type references to own class
+# https://stackoverflow.com/a/33533514
+from __future__ import annotations
+from typing import List
 import datetime
 import time
-
 import pytz
 from asgiref.sync import sync_to_async
 from django.conf import settings
 from django.db import models
 from django.forms import model_to_dict
+from .customFields import FixedCharField, GeneratedIdentityField
+
+
+import logging
+logger = logging.getLogger('wall_e')
+
+
+class BanRecords(models.Model):
+    ban_id = GeneratedIdentityField(primary_key=True, always=True)
+    username = models.CharField(max_length=32, null=False)
+    user_id = models.BigIntegerField(null=False)
+    mod = models.CharField(max_length=32, null=True)
+    mod_id = models.BigIntegerField(null=True)
+    ban_date = models.BigIntegerField(null=True)
+    reason = models.TextField(null=False)
+    unban_date = models.BigIntegerField(null=True, default=None)
+
+    class Meta:
+        db_table = 'WalleModels_ban_records'
+
+    @classmethod
+    @sync_to_async
+    def insert_records(cls, records: List[BanRecords]) -> None:
+        """Adds entry to BanRecords table"""
+        for record in records:
+            record.save()
+
+    @classmethod
+    @sync_to_async
+    def insert_record(cls, record: BanRecords) -> None:
+        """Adds entry to BanRecords table"""
+        record.save()
+
+    @classmethod
+    @sync_to_async
+    def get_all_active_ban_user_ids(cls) -> List[int]:
+        """Returns list of user_ids for all currently banned users"""
+
+        return list(BanRecords.objects.values_list('user_id', flat=True).filter(unban_date=None))
```

Comment:
please use named tuples here and for `get_all_active_bans`. I know that there's a bit of a performance hit to doing so but I think its an acceptable trade-off for not having to do indexing.

Reply:
k, but for the `get_all_active_ban_user_ids` functions it's not needed, its only used in 1 place the list is just used to the the ban list without any indexing. ill make the change in `get_all_active_bans`.

Reply:
yes, sorry. i forgot about that.